src/cljs/athens/events.cljs

```diff
@@ -486,9 +486,11 @@
                        db/get-block)
         new-block {:db/id (:db/id block) :block/order (count (:block/children older-sib))}
         reindex-blocks (->> (dec-after (:db/id parent) (:block/order block)))]
-    {:transact! [[:db/retract (:db/id parent) :block/children (:db/id block)]
-                 {:db/id (:db/id older-sib) :block/children [new-block]} ;; becomes child of older sibling block — same parent but order-1
-                 {:db/id (:db/id parent) :block/children reindex-blocks}]}))
+    (if (zero? (:block/order block)) ; If the block is already the top in it's level, do not indent
```

Comment:
Since this condition is context-less — it doesn't require any re-frame or datascript subscriptions — I would add it directly to the giant `on-key-down` handler.

Reply:
Moving the condition to the `on-key-down` handler has the side effect of not reaching the `preventDefault` code - 

```clojure
(and (= key-code KeyCodes.TAB) (not block-zero?)) (do (.. e preventDefault)
                                                            (dispatch [:indent uid]))
```